servers/src/test/java/tachyon/master/next/filesystem/meta/InodeTreeTest.java

```diff
@@ -0,0 +1,208 @@
+    mThrown.expectMessage("Could not traverse to parent directory of path /nested/test/test."
+        + " Component test is not a directory.");
+
+    mTree.createPath(NESTED_URI, Constants.KB, true, false);
+    mTree.createPath(new TachyonURI("/nested/test/test"), Constants.KB, true, false);
+  }
+
+  @Test
+  public void getInodeByNonexistingPathTest() throws Exception {
+    mThrown.expect(InvalidPathException.class);
+    mThrown.expectMessage("Could not find path: /test");
+
+    mTree.getInodeByPath(TEST_URI);
+  }
+
+  @Test
+  public void getInodeByInvalidIdTest() throws Exception {
+    mThrown.expect(FileDoesNotExistException.class);
+    mThrown.expectMessage("Inode id 1 does not exist");
+
+    mTree.getInodeById(1);
+  }
+
+  @Test
+  public void isRootIdTest() throws Exception {
+    Assert.assertTrue(mTree.isRootId(0));
+    Assert.assertFalse(mTree.isRootId(1));
+  }
+
+  @Test
+  public void getPathTest() throws Exception {
+    Inode root = mTree.getInodeById(0);
+    // test root path
+    Assert.assertEquals(new TachyonURI("/"), mTree.getPath(root));
+
+    // test one level
+    Inode test = mTree.createPath(TEST_URI, Constants.KB, false, true);
+    Assert.assertEquals(new TachyonURI("/test"), mTree.getPath(test));
+
+    // test nesting
+    Inode nested = mTree.createPath(NESTED_URI, Constants.KB, true, true);
+    Assert.assertEquals(new TachyonURI("/nested/test"), mTree.getPath(nested));
+  }
+
+  @Test
+  public void getInodeChildrenRecursiveTest() throws Exception {
+    mTree.createPath(TEST_URI, Constants.KB, false, true);
+    mTree.createPath(NESTED_URI, Constants.KB, true, true);
+    // add nested file
+    mTree.createPath(NESTED_FILE_URI, Constants.KB, true, false);
+
+    // all inodes under root
+    List<Inode> inodes = mTree.getInodeChildrenRecursive((InodeDirectory) mTree.getInodeById(0));
+    // /test, /nested, /nested/test, /nested/test/file
+    Assert.assertEquals(4, inodes.size());
+  }
+
+  @Test
+  public void deleteInodeTest() throws Exception {
+    Inode nested = mTree.createPath(NESTED_URI, Constants.KB, true, true);
+
+    // all inodes under root
+    List<Inode> inodes = mTree.getInodeChildrenRecursive((InodeDirectory) mTree.getInodeById(0));
+    // /nested, /nested/test
+    Assert.assertEquals(2, inodes.size());
+    // delete the nested inode
+    mTree.deleteInode(nested);
+    inodes = mTree.getInodeChildrenRecursive((InodeDirectory) mTree.getInodeById(0));
+    // only /nested left
+    Assert.assertEquals(1, inodes.size());
+  }
+
+  @Test
+  public void deleteNonexistingInodeTest() throws Exception {
+    mThrown.expect(FileDoesNotExistException.class);
+    mThrown.expectMessage("Inode id 1 does not exist");
+
+    Inode testFile = new InodeFile("testFile1", 1, 1, Constants.KB, System.currentTimeMillis());
+    mTree.deleteInode(testFile);
+  }
+
+  @Test
+  public void setPinnedTest() throws Exception {
+    Inode nested = mTree.createPath(NESTED_URI, Constants.KB, true, true);
```

Comment:
What about putting these `mTree.createPath` into helper functions like `createNestedUri()`, `createNestedFileUri()`...  they could be helper for future tests because determining each parameter in `createPath` seems not intuitive, especially the `true/false`.